lib/settings/use-saved-settings.js

```diff
@@ -31,13 +31,13 @@ export default (settingsId, settings, setSettings, onReset) => {
 		onReset: useCallback(
 			async (e) => {
 				setSettings();
+				setSavedSettings();
 				if (e.shiftKey) {
-					await write(settingsId);
-					setSavedSettings();
+					await write(settingsId, {});
 				}
 				onReset?.();
 			},
-			[onReset]
+			[settings, savedSettings, write, settingsId],
```

Comment:
`settings` and `savedSettings` are not dependencies of this callback, but `onReset` is.

```suggestion
			[onReset, write, settingsId],
```